src/modules/flow/iio/nodes.c

```diff
@@ -959,6 +959,178 @@ light_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_op
 }
 
 static bool
+intensity_both_create_channels(struct iio_double_data *mdata, int device_id, const int channel_id)
+{
+    char channel_name[NAME_MAX];
+    int ret = 0;
+
+    mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
+    SOL_NULL_CHECK(mdata->iio_base.device, false);
+
+    if (channel_id >= 0) {
+        ret = snprintf(channel_name, sizeof(channel_name), "in_intensity%d_both", channel_id);
+        SOL_INT_CHECK_GOTO(ret, >= (int)sizeof(channel_name), error);
+        SOL_INT_CHECK_GOTO(ret, < 0, error);
+    } else {
+        strncpy(channel_name, "in_intensity_both", sizeof(channel_name) - 1);
```

Comment:
Hint: as you are copying a constant into buffer, no strong need for `strncpy` and '\0' worries. But it doesn't hurt, either.